contracts/exchangeIssuance/DEXAdapter.sol

```diff
@@ -0,0 +1,340 @@
+        uint256 _maxAmountIn
+    )
+    internal
+    returns(uint256)
+    {
+
+        require(_path.length == _fees.length + 1, "ExchangeIssuance: PATHS_FEES_MISMATCH");
+        _safeApprove(IERC20(_path[0]), address(uniV3Router), _maxAmountIn);
+        if(_path.length == 2){
+            ISwapRouter.ExactOutputSingleParams memory params =
+                ISwapRouter.ExactOutputSingleParams({
+                    tokenIn: _path[0],
+                    tokenOut: _path[1],
+                    fee: _fees[0],
+                    recipient: address(this),
+                    deadline: block.timestamp,
+                    amountOut: _amountOut,
+                    amountInMaximum: _maxAmountIn,
+                    sqrtPriceLimitX96: 0
+                });
+
+            // Executes the swap returning the amountIn needed to spend to receive the desired amountOut.
+            return uniV3Router.exactOutputSingle(params);
+        } else {
+            bytes memory pathV3 = _encodePathV3(_path, _fees, true);
+            uint outputBalanceBefore = IERC20(_path[_path.length - 1]).balanceOf(address(this));
+            ISwapRouter.ExactOutputParams memory params =
+                ISwapRouter.ExactOutputParams({
+                    path: pathV3,
+                    recipient: address(this),
+                    deadline: block.timestamp,
+                    amountOut: _amountOut,
+                    amountInMaximum: _maxAmountIn
+                });
+            uint amountIn = uniV3Router.exactOutput(params);
+            uint outputBalanceAfter = IERC20(_path[_path.length - 1]).balanceOf(address(this));
+            uint actualAmountOut = outputBalanceAfter.sub(outputBalanceBefore);
+            return amountIn;
+        }
+    }
+
+    /**
+     *  Execute exact input swap via UniswapV3
+     *
+     * @param _path         List of token address to swap via. 
+     * @param _fees         List of fee levels identifying the pools to swap via. (_fees[0] refers to pool between _path[0] and _path[1])
+     * @param _amountIn    The amount of input token to be spent
+     *
+     * @return amountOut    The amount of output token obtained
+     */
+    function _swapExactTokensForTokensUniV3(
+        address[] memory _path,
+        uint24[] memory _fees,
+        uint256 _amountIn
+    )
+    internal
+    returns(uint256)
+    {
+        require(_path.length == _fees.length + 1, "ExchangeIssuance: PATHS_FEES_MISMATCH");
+        _safeApprove(IERC20(_path[0]), address(uniV3Router), _amountIn);
+        if(_path.length == 2){
+            ISwapRouter.ExactInputSingleParams memory params =
+                ISwapRouter.ExactInputSingleParams({
+                    tokenIn: _path[0],
+                    tokenOut: _path[1],
+                    fee: _fees[0],
+                    recipient: address(this),
+                    deadline: block.timestamp,
+                    amountIn: _amountIn,
+                    amountOutMinimum: 0,
```

Comment:
There is no good reason for the stated inconsistency regarding slippage checks. I agree there should not be any sandwich risk here since we ensure later that the caller does receive at least the specified output amount. 

I will remove the slippage check in the other function and document as you suggested.

Reply:
Good point regarding the interface. Will adjust to make functions intended for use inside this contract private.

Reply:
done